API_doc/Communication.md

```diff
@@ -4,14 +4,20 @@
 
 All requests must contain **REQUEST_NAME**, containing the type of action they want to do.
 Each response contain at least **REQUEST_STATE** (see below).
+All the local settings are applied after the config inclusions.
 
 ### Request types
 | Request Name | Request Description | Additional Argument(s) | Additional Returned Value(s) |
 | -------- | -------- | -------- | -------- |
 |*CONFIG_CREATE*|Create a new config.|**CONFIG_NAME**|**CONFIG_KEY**<br>**READONLY_CONFIG_KEY**|
 |*CONFIG_LOAD*| Load an existing config. |**CONFIG_KEY** *or* **READONLY_CONFIG_KEY**| **CONFIG_NAME**<br>**CONFIG_ID**|
 |*CONFIG_UNLOAD*| Unload config. |**CONFIG_ID**|*none*|
-|*CONFIG_INCLUDE*| Include a config in another |**CONFIG_ID**<br>**SRC**<br>**DEST**|*none*|
+|*CONFIG_DELETE*| Delete config. |**CONFIG_ID**|*none*|
+|*CONFIG_GET_DEPS*| Get the ordered list of all included configs. |**CONFIG_ID**|**DEPS** (array of READONLY_CONFIG_KEY)|
```

Comment:
I would maybe add a CONFIG_GET_LOADED_OR_LOAD_DEPS, which would load all not yet loaded configs and return config_ids. The thing is I can imagine many scenarios where the client doesn't care about the key itself.

Reply:
If we do that, we also have to make a "UNLOAD_DEPS" to call "CONFIG_UNLOAD" for each dependency

Reply:
I feel like 'CONFIG_LOAD' should always load the deps ?

Reply:
No, I believe keeping the effect of config load simple (just loading the config itself) is the best way to  go. I don't think it's necessary to provide "UNLOAD_DEPS". The user can just iterate and call "CONFIG_UNLOAD" for each config_id.

Reply:
I found that weird to force the user to call "CONFIG_UNLOAD" for configs that he didn't load himself.

Reply:
It's true that unless all configs are loaded again, it's hard for the user to know which configs to free.
If all configs are assigned new ids, he'll have a hard time knowing who's who.
The only solution I see is introducing config reference counting inside the library.
The library would make only one CONFIG_LOAD request and keep an internal counter for the amount of times a config is loaded beyond that and return the same id. When CONFIG_UNLOAD is called, the reference count would be decreased and the library would only do a CONFIG_UNLOAD request when the reference count reaches 0.
This implies this CONFIG_GET_LOADED_OR_LOAD_DEPS would return an additional value for each config specifying if this id is already loaded, and the library would create a new ref count or increase the old one accordingly. This seems overall quite a task, and I can understand if we want to do this at a later point in time.

Reply:
We need to talk about that because I don't understand all

Reply:
need to make the unload with array